core/src/main/java/org/commonjava/indy/core/ctl/AdminController.java

```diff
@@ -151,11 +168,27 @@ public ArtifactStore get( final StoreKey key )
         }
     }
 
-    public void delete( final StoreKey key, final String user, final String changelog )
+    public void delete( final StoreKey key, final String user, final String changelog, final boolean deleteContent )
         throws IndyWorkflowException
     {
+        // safe check
```

Comment:
Would it make more sense to delete the content after the store definition is deleted? That way the content would still be recoverable in case the store deletion failed?

Reply:
After repo definition being deleted, the content deletion will fail because the store is missing.

Reply:
ugh, ok